File: Proyecto/Calendario_App/src/main/java/org/example/calendario_app/dao/ClienteDAO.java
package org.example.calendario_app.dao;

import org.example.calendario_app.dao.impl.ClienteDAOImpl;
import org.example.calendario_app.model.Cliente;

public class ClienteDAO {
    ClienteDAOImpl clienteDAO;

    public ClienteDAO(ClienteDAOImpl clienteDAO) {
        this.clienteDAO = clienteDAO;
    }

    public int registrar(Cliente cliente) {
        return clienteDAO.registrar(cliente);
    }

    public Cliente obtenerPorId(int id) {
        return clienteDAO.obtenerPorId(id);
    }
}

File: Proyecto_Android/Calendario_Android_App-31-01-2026/Calendario_Android_App-31-01-2026/Calendario_Android_App/app/src/main/java/com/example/calendario_android_app/dao/ClienteDAO.kt
package com.example.calendario_android_app.dao

import com.example.calendario_android_app.model.Cliente

/**
 * Interfaz que define las operaciones de acceso a datos para la entidad Cliente.
 * Los clientes representan personas físicas que pueden tener cuentas de usuario.
 */
interface ClienteDAO {
    /**
     * Obtiene un cliente por su identificador.
     * @param id_cliente ID del cliente a buscar
     * @return Cliente encontrado o null si no existe
     */
    fun obtenerCliente(id_cliente: Int): Cliente?
    
    /**
     * Crea un nuevo cliente en la base de datos.
     * @param nombre Nombre del cliente
     * @param apellidos Apellidos del cliente
     * @return ID del cliente creado o null si falla la operación
     */
    fun crearCliente(nombre: String, apellidos: String): Int?
}